src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentDesignTimeNodeWriter.cs

```diff
@@ -596,6 +622,90 @@ public override void WriteComponentAttribute(CodeRenderingContext context, Compo
         context.CodeWriter.WriteLine();
     }
 
+    private void WritePropertyAccess(CodeRenderingContext context, ComponentAttributeIntermediateNode node, ComponentIntermediateNode componentNode, string typeInferenceLocalName)
+    {
+        if (node?.TagHelper?.Name is null || node.Annotations["OriginalAttributeSpan"] is null)
+        {
+            return;
+        }
+
+        // Write the name of the property, for rename support. 
+        // __o = nameof(global::ComponentName.PropertyName);
+        var originalAttributeName = node.Annotations[ComponentMetadata.Common.OriginalAttributeName]?.ToString() ?? node.AttributeName;
+
+        int offset;
+        if (originalAttributeName == node.PropertyName)
+        {
+            offset = 0;
+        }
+        else if (originalAttributeName.StartsWith($"@bind-{node.PropertyName}", StringComparison.Ordinal))
+        {
+            offset = 5;
+        }
+        else
+        {
+            return;
+        }
+
+        var attributeSourceSpan = (SourceSpan)node.Annotations["OriginalAttributeSpan"];
+        attributeSourceSpan = new SourceSpan(attributeSourceSpan.FilePath, attributeSourceSpan.AbsoluteIndex + offset, attributeSourceSpan.LineIndex, attributeSourceSpan.CharacterIndex + offset, node.PropertyName.Length, attributeSourceSpan.LineCount, attributeSourceSpan.CharacterIndex + offset + node.PropertyName.Length);
+
+        context.CodeWriter.Write(DesignTimeVariable);
+        context.CodeWriter.Write(" = ");
+        context.CodeWriter.Write("nameof(");
+
+        if (componentNode.TypeInferenceNode == null)
+        {
+            context.CodeWriter.Write("global::");
```

Comment:
Are the symbol types here limited? Are we sure that `global::` will always be a valid prefix?

Reply:
I don't know when `global::` isn't valid (but happy to be educated) but I couldn't see any consideration given in the `__o = typeof(global::...)` codegen that appears just above this.

Reply:
> I couldn't see any consideration given in the __o = typeof(global::...) codegen that appears just above this.

That is definitely not a good metric, we've had to fix multiple scenarios where `global` is being put in front of something that it can't be since taking over the codebase.

Consider scenarios like:
* Can this type be a keyword such as `string` or `dynamic`?
* Can this type be a tuple?